### ISSUE TITLE
Updb fails on fresh Lightning installs with Drupal 8.4

### ISSUE THREAD

**Issue by danepowell** [acquia/lightning#496]
Install a site using Lightning 2.2.1 and Drupal core 8.4, and then after the fresh install try to run Drush updb, and you will be greeted by this lovely and cryptic error:

<blockquote>Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: You have requested a non-existent service "plugin.manager.media_entity.type".</blockquote>

The root cause seems to be the service call in this patch:
https://www.drupal.org/files/issues/media-entity-1.6--c564fc4.diff

You can see this in the wild with this failing BLT test:
https://travis-ci.org/acquia/blt/builds/292900603

**Comment by balsama** [comment]
Thanks. I don't think we test what happens when you run updb on a fresh install, so this is totally plausible.

Right now, we're working on fixing all of the hook requirement problems which we were utterly unaware of until Drush fixed updatedb to actually check hook_requirements in 9.0.0-beta8. Will look into this right after. Thanks.

**Comment by balsama** [comment]
@danepowell Media Entity shouldn't be installed on a fresh install of 2.2.1 (and the update hooks should uninstall it for existing sites).

Do you have some other dependency that still depends on Media Entity (and is installing it) perhaps?

**Comment by danepowell** [comment]
You're right, Media Entity is not installed. And yet, the error :confused:

**Comment by danepowell** [comment]
I wonder if Drush and/or Drupal core processes hook_requirements() at update time even for uninstalled modules. I can't imagine that's the case, but it sure seems like it.

**Comment by danepowell** [comment]
@balsama oh I think I understand now, it's not hook_requirements making the call, it's  media_entity.drush.inc, line 39:

```php
/**
 * Implements drush_hook_COMMAND_validate().
 */
function drush_media_entity_updatedb_validate() {
// snip
$incompatible_modules = _media_entity_get_incompatible_modules();
```

**Comment by balsama** [comment]
OK. So we fixed a bunch of undetected hook_requirement issues here: https://github.com/acquia/lightning/commit/61660cb55b6a5b6b94eb65d126d439f0f05a597a

What I don't understand is why that hook would be loaded if Media Entity is uninstalled. I'd like to see if this is fixed in HEAD of Lightning's 8.x-2.x.

**Comment by danepowell** [comment]
@balsama so I'm having a little trouble testing this.

I can't get a site using Drush 8 to upgrade past Lightning 2.2.1. I suspect this is because of how Lightning now pins the Drush version to 9.0.0-beta7, but that doesn't totally make sense because it's in require-dev. It _could_ be some sort of transient packagist issue (I've already tried clearing caches). This in itself is going to be a larger problem if it persists.

I also can't reproduce this on a site running Drush 9. My guess is that Drush 9 no longer runs that hook in media_entity.drush.inc. Which fixes this "bug", but in itself is also probably a larger problem :)

**Comment by balsama** [comment]
Ok. I'm reproducing this now. Still not sure what the minimum steps to reproduce are since I added a cache rebuild and db update to the end of our internal test script.

But I'm able to reproduce on an existing BLT project.

**Comment by balsama** [comment]
Alright. So the patch here finally makes sense after seeing Drush load that hook when the module is *disabled*. https://www.drupal.org/node/2919754#comment-12322255

I'm not 100% certain which versions of Drush have that bug, but rather than try to fix it in Drush, I concur with the patch and let's just bail out of the module is uninstalled.

Todo: add that patch to Lightning's composer.json.

**Comment by balsama** [comment]
Fixed in 8913bd764b2b877946a7d290c140a49808f7e89a via https://www.drupal.org/node/2919754#comment-12322619

**Comment by danepowell** [comment]
Confirmed, this is fixed in 2.2.2-alpha2, thanks!
